### PR TITLE
[TE] rootcause poc - add user-readable alerts on data error

### PR DESCRIPTION
**Pull request by apucher**
The UI currently does not surface data retrieval errors to the user. This PR adds basic user-readable error message and enables message dismissal.

image

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js [hunk 1]
```diff
@@ -257,6 +257,19 @@ export default Ember.Controller.extend({
     }
   ),
 
+  hasErrors: Ember.computed(
+    'entitiesService.errors',
+    'timeseriesService.errors',
+    'aggregatesService.errors',
+    'breakdownsService.errors',
+    function () {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: please use 
```
Ember.computed.or('entitiesService.errors', 'timeseriesService.errors','aggregatesService.errors', 'breakdownsService.errors')
```

https://emberjs.com/api/ember/2.17/classes/@ember%2Fobject%2Fcomputed/methods/or?anchor=or

^You may pass in more than two properties and even use property brace expansion. The computed property will return the first truthy value or last falsy value just like JavaScript's || operator

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+    <a class="close" data-dismiss="alert" {{action "clearErrors"}}>&times;</a>
+    <ul>
+      {{#if entitiesService.errors.size}}
+        <li>Error for task(s):{{#each entitiesService.errors as |e|}} <strong>{{e.framework}}</strong>{{/each}}</li>
+      {{/if}}
+      {{#if timeseriesService.errors.size}}
+        <li>Could not load timeseries:{{#each timeseriesService.errors as |e|}} <strong>{{e.urn}}</strong>{{/each}}</li>
+      {{/if}}
+      {{#if aggregatesService.errors.size}}
+      <li>Could not load aggregates:{{#each aggregatesService.errors as |e|}} <strong>{{e.urn}}</strong>{{/each}}</li>
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: nit: indent

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+      {{/if}}
+      {{#if breakdownsService.errors.size}}
+      <li>Could not load dimension:{{#each breakdownsService.errors as |e|}} <strong>{{e.urn}}</strong>{{/each}}</li>
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: same

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs [hunk 4]
```diff
@@ -1,3 +1,23 @@
+{{#if hasErrors}}
+  <div class="alert alert-danger alert-dismissable fade in">
+    <a class="close" data-dismiss="alert" {{action "clearErrors"}}>&times;</a>
+    <ul>
+      {{#if entitiesService.errors.size}}
+        <li>Error for task(s):{{#each entitiesService.errors as |e|}} <strong>{{e.framework}}</strong>{{/each}}</li>
+      {{/if}}
+      {{#if timeseriesService.errors.size}}
+        <li>Could not load timeseries:{{#each timeseriesService.errors as |e|}} <strong>{{e.urn}}</strong>{{/each}}</li>
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: I can see a pattern here, perhaps an opportunity to componentized later on (obviously we'll need a design first :) )

File: thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js (hunk 1 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: fixed

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs (hunk 2 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: fixed

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs (hunk 3 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: fixed

**Pull request merged**